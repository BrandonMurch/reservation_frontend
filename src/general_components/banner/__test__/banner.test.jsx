import React from 'react';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import Banner, { bannerTypes } from '../index';

describe('<Banner />', () => {
  let component;
  let container = null;
  beforeEach(() => {
    component = render(<Banner
      type={bannerTypes.SUCCESS}
      message="banner message"
    />);
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  it('should match snapshot', () => {
    const tree = renderer.create(<Banner
      type={bannerTypes.SUCCESS}
      message="banner message"
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should display a message', () => {
    const element = component.getByText(/banner message/i);
    expect(element).toBeInTheDocument();
  });
  it('should be have success classname', () => {
    const banner = component.getByRole('banner');
    expect(banner.className).toEqual('success');
  });
});
