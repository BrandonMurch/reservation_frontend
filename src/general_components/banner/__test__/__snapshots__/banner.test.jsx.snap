// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should match snapshot 1`] = `
<div
  className="success"
  role="banner"
>
  <p
    style={
      Object {
        "whiteSpace": "break-spaces",
      }
    }
  >
    banner message
  </p>
</div>
`;
